lib/pymedphys/tests/dicom/test_connect.py

```diff
@@ -295,3 +297,131 @@ def test_dicom_listener_cli(test_dataset):
         file_path = _build_hierarchical_path_to_plan(test_directory, test_dataset)
         read_dataset = pydicom.read_file(file_path)
         assert read_dataset.SeriesInstanceUID == test_dataset.SeriesInstanceUID
+
+
+@pytest.mark.pydicom
+def test_dicom_sender(test_dataset):
+    """Test sending DICOM objects using the DicomSender"""
+
+    scp_ae_title = "PYMEDPHYSTEST"
+
+    with tempfile.TemporaryDirectory() as tmp_directory:
+
+        test_directory = pathlib.Path(tmp_directory)
+        receive_directory = test_directory.joinpath("receive")
+        receive_directory.mkdir()
+
+        listener_command = [
+            pmp_test_utils.get_executable_even_when_embedded(),
+            "-m",
+            "pymedphys",
+            "dicom",
+            "listen",
+            str(TEST_PORT),
+            "-d",
+            str(receive_directory),
+            "-a",
+            str(scp_ae_title),
+        ]
+
+        with process(listener_command):
+            time.sleep(1)
+            dicom_sender = DicomSender(
+                host="127.0.0.1", port=TEST_PORT, ae_title=scp_ae_title
+            )
+            assert dicom_sender.verify()
+            dicom_sender.send([test_dataset])
+
+        dcm_files = receive_directory.glob("**/*.dcm")
+        dcm_file = next(iter(dcm_files))
+        ds = pydicom.read_file(dcm_file)
+        assert ds.SOPInstanceUID == test_dataset.SOPInstanceUID
+        assert ds.SeriesInstanceUID == test_dataset.SeriesInstanceUID
+        assert ds.StudyInstanceUID == test_dataset.StudyInstanceUID
+        assert ds.PatientID == test_dataset.PatientID
+        assert ds.Modality == test_dataset.Modality
+        assert ds.Manufacturer == test_dataset.Manufacturer
+
+        assert len(ds.BeamSequence) == 1
+        assert (
+            ds.BeamSequence[0].Manufacturer == test_dataset.BeamSequence[0].Manufacturer
+        )
+
+
+@pytest.mark.pydicom
+def test_dicom_sender_cli(test_dataset):
+    """Test the command line interface to the DicomSender"""
+
+    scp_ae_title = "PYMEDPHYSTEST"
+
+    with tempfile.TemporaryDirectory() as tmp_directory:
+
+        test_directory = pathlib.Path(tmp_directory)
+        send_directory = test_directory.joinpath("send")
+        send_directory.mkdir()
+        send_file = send_directory.joinpath("test.dcm")
+        test_dataset.save_as(send_file)
+
+        # Need to write the preamble or else we pydicom throws an error when reading the
+        # file... This solution is from: https://github.com/pydicom/pydicom/issues/340
+        # TODO: Not entirely sure why this is needed. Is there a cleaner solution to
+        # this?
```

Comment:
`dcmwrite` has utilities to automatically handle the preamble:

https://pydicom.github.io/pydicom/dev/reference/generated/pydicom.filewriter.dcmwrite.html